dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java

```diff
@@ -58,63 +57,65 @@
             throws DotSecurityException, DotDataException {
 
-        if (this.matchingUrlPattern(context.getUri())) {
-            final Matches matches = this.findPatternChange(context.getUri());
-
-            final Structure structure = CacheLocator.getContentTypeCache()
-                    .getStructureByInode(matches.getPatternChange().getStructureInode());
-
-            final Field hostField = this.findHostField(structure);
-
-            final Contentlet contentlet = this.getContentlet(matches, structure, hostField, context);
-
-            if (contentlet == null) {
-                return Optional.empty();
-            }
-
-            final Identifier pageUriIdentifier = this.getDetailtPageUri(structure);
-
-            return Optional.of(new URLMapInfo(contentlet, pageUriIdentifier, context.getUri()));
-        } else {
+        final Contentlet contentlet = getContentlet(context);
+        if (contentlet == null) {
             return Optional.empty();
         }
+
+        final Structure structure = CacheLocator.getContentTypeCache()
+                .getStructureByInode(contentlet.getStructureInode());
+        final Identifier pageUriIdentifier = this.getDetailtPageUri(structure);
+
+        return Optional.of(new URLMapInfo(contentlet, pageUriIdentifier, context.getUri()));
     }
 
     /**
-     * Return the {@link Contentlet} the match the {@link UrlMapContext#getUri()} value,
-     * if not exists any {@link com.dotcms.contenttype.model.type.UrlMapable} matching with the URI
-     * then a {@link DotRuntimeException} is thrown
+     * Return the {@link Contentlet} the match the {@link UrlMapContext#getUri()} value, if not
+     * exists any {@link com.dotcms.contenttype.model.type.UrlMapable} matching with the URI then a
+     * {@link DotRuntimeException} is thrown
      *
      * @param urlMapContext
      * @return
      */
-    private Contentlet getContentlet(final UrlMapContext urlMapContext){
-        final Matches matches = this.findPatternChange(urlMapContext.getUri());
-        final Structure structure = CacheLocator.getContentTypeCache()
-                .getStructureByInode(matches.getPatternChange().getStructureInode());
+    private Contentlet getContentlet(final UrlMapContext urlMapContext) throws DotSecurityException {
 
-        final Field hostField = this.findHostField(structure);
+        Contentlet matchingContentlet = null;
 
         try {
-            return this.getContentlet(matches, structure, hostField, urlMapContext);
-        } catch (DotDataException | DotSecurityException e){
-            return null;
-        }
-    }
+            // We could have multiple matches as multiple content types could have the same
+            // URLMap pattern and we need to evaluate all until we find content match.
+            final List<Matches> matchesFound = this.findMatch(urlMapContext.getUri());
+            if (!matchesFound.isEmpty()) {
+
+                for (Matches matches : matchesFound) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'matches' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5176032)